spec/Appendix A -- Field Selection.md

```diff
@@ -0,0 +1,797 @@
+Specifically, it allows defining complex relationships between input arguments and fields in the output object by encapsulating these relationships within a parsable string format. 
+It is used in the `@is` and `@require` directives.
+
+To illustrate, consider a simple example from a GraphQL schema:
+
+```graphql
+type Query {
+    userById(userId: ID! @is(field: "id")): User! @lookup
+}
+```
+
+In this schema, the `userById` query uses the `@is` directive with {FieldSelectionMap} to declare that the `userId` argument is semantically equivalent to the `User.id` field. 
+
+An example query might look like this:
+
+```graphql
+query {
+    userById(userId: "123") {
+        id
+    }
+}
+```
+
+Here, it is exptected that the `userId` "123" corresponds directly to `User.id`, resulting in the following response if correctly implemented:
+
+```json
+{
+    "data": {
+        "userById": {
+            "id": "123"
+        }
+    }
+}
+```
+
+The {FieldSelectionMap} scalar is represented as a string that, when parsed, produces a {SelectedValue}.
+
+A {SelectedValue} must exactly match the shape of the argument value to be considered valid. For non-scalar arguments, you must specify each field of the input type in {SelectedObjectValue}.
+
+```graphql example
+extend type Query {
+    findUserByName(user: UserInput! @is(field: "{ firstName: firstName }")): User @lookup
+}
+```
+
+```graphql counter-example
+extend type Query {
+    findUserByName(user: UserInput! @is(field: "firstName")): User @lookup
+}
+```
+
+### Scope
+The {FieldSelectionMap} scalar type is used to establish semantic equivalence between an argument and fields within a specific output type.
+This output type is always a composite type, but the way it's determined can vary depending on the directive and context in which the {FieldSelectionMap} is used.
+
+For example, when used with the `@is` directive, the {FieldSelectionMap} maps between the argument and fields in the return type of the field. 
+However, when used with the `@require` directive, it maps between the argument and fields in the object type on which the field is defined.
+
+Consider this example:
+
+```graphql
+type Product {
+  id: ID!
+  delivery(
+    zip: String!
+    size: Int! @require(field: "dimension.size")
+    weight: Int! @require(field: "dimension.weight")
+  ): DeliveryEstimates
+}
+```
+
+In this case, `"dimension.size"` and `"dimension.weight"` refer to fields of the `Product` type, not the `DeliveryEstimates` return type.
+
+Consequently, a {FieldSelectionMap} must be interpreted in the context of a specific argument, its associated directive, and the relevant output type as determined by that directive's behavior.
+
+**Examples**
+
+Scalar fields can be mapped directly to arguments.
+
+This example maps the `Product.weight` field to the `weight` argument:
+
+```graphql example
+type Product {
+    shippingCost(weight: Float @require(field: "weight")): Currency
```

Comment:
just wondering -> with all those examples would it make simpler to grasp if the other types (well their stubs) were present? (this gets somewhat confusing once you get to the complex mapping of top level field and nested fields)

*also those other types are included when talking about the syntax

Reply:
Well it's a big confusing if we add 'weight' here, as this would represent the already "merged" type from the composed schema instead of a type on a source schema. I think we should leave it out here.